### ISSUE TITLE
Allow aws_organizations_account resource creation to block until new account is usable

### ISSUE THREAD

**Issue by mariadb-JeffBachtel** [hashicorp/terraform-provider-aws#15654]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

A failing of the `organizations:CreateAccount` API call is that an account can be created, but it might not yet be possible to use the account for 30 seconds or more. It would be keen if there was an option in the `aws_organizations_account` resource to block until the generated account was verifiably useful.

### New or Affected Resource(s)

* `aws_organizations_account`

### Potential Terraform Configuration

```hcl
resource "aws_organizations_account" "new_account" {
  name     = var.new_account_name
  email    = var.new_account_email
  wait_for_ready = true
}
```

### References

In an example where a Route53 zone was specified in an account newly created in the same terraform template:

```
2020-10-14 22:36:58,305 ERROR:
Error: error creating Route53 Hosted Zone: OptInRequired: The AWS Access Key Id needs a subscription for the service
	status code: 403, request id: 0a0cab20-1cd7-4729-8c95-8216243e07c4
```

When `terraform apply` was rerun about 30 seconds later, the R53 provisioning succeeded

**Comment by ChristophShyper** [comment]
There are API calls to check if the account it ready: https://docs.aws.amazon.com/cli/latest/reference/organizations/describe-create-account-status.html
So provider can call it every few seconds. I'm now implementing additional script to use with `local-exec`, because recently AWS takes even 3 minutes to prepare the account.

**Comment by mariadb-JeffBachtel** [comment]
Unfortunately, `organizations:DescribeCreateAccountStatus` (which the aws provider uses) does not cover whether Route53 and S3 have been provisioned into an account. It's a weird AWS-ism.

**Comment by ChristophShyper** [comment]
Then using https://registry.terraform.io/providers/hashicorp/time/latest/docs/resources/sleep is the only solution.
Few months ago 1 minute was enough (I used local-exec back then), now it's around 3-5 minutes.

**Comment by mariadb-JeffBachtel** [comment]
Um well no, this ticket is to get a better solution added to the provider.

So, one thought I had was a loop to create a randomly named S3 bucket, sleeping between runs, with a max timeout. And then an immediate deletion. Similar with a R53 zone. All of that could be behind an attribute `wait_for_services` or similar, so that the current behavior is the default.

**Comment by slmagus** [comment]
If aws were to enhance or create a new api what sort of information should it provide?

**Comment by drewmullen** [comment]
@mariadb-JeffBachtel  could you use https://registry.terraform.io/providers/hashicorp/time/latest/docs/resources/sleep ?

**Comment by ChristophShyper** [comment]
@slmagus It should allow to check if all default resources are ready.

Recently account creation took 7 minutes... 
Fortunately we're not creating account daily, so I've increased timeout to 10 minutes now.

**Comment by mariadb-JeffBachtel** [comment]
@drewmullen no, because probes need to be done at a provider level to determine whether various AWS APIs are ready to go. You cannot realistically tell when that will be. Sometimes it's a minute, sometimes 7 minutes.

@slmagus an AWS API to enumerate the services enabled for an account (S3 and R53 being the only ones I've ever seen this issue apply to) would suffice for the provider to query, to avoid returning until the account creation had completed

**Comment by 1davidmichael** [comment]
> @drewmullen no, because probes need to be done at a provider level to determine whether various AWS APIs are ready to go. You cannot realistically tell when that will be. Sometimes it's a minute, sometimes 7 minutes.
> 
> @slmagus an AWS API to enumerate the services enabled for an account (S3 and R53 being the only ones I've ever seen this issue apply to) would suffice for the provider to query, to avoid returning until the account creation had completed

I've encountered this issue with KMS as well.

**Comment by ChristophShyper** [comment]
Cool it's closed without any valid explanation.... `/sarcasm`

Recently even 7 minutes timeout is not enough for the the provider to handle resource creation properly. 
For example on of the places are the default tags in the provider, which fail when applying just after 7minutes after the account creation. But on the second run (around 1-2 minutes after they pass properly). 


As a workaround I use a semaphore variable in Terragrunt to use jinja to enable default tags after I manually switch one variable (e.g. after enabling ChatBot for Slack).
But it's a shame that in 2025 we need to find suitable timeouts and triggers by chance instead of just have a working code... All feels like working with APIs 20-30 years ago...

**Comment by drewmullen** [comment]
Bad news based on my initial look. It does look like the provider is already referencing that specific API call and waiting for AWS to report the account as "SUCCEEDED". This could actually be a bug at AWS API layer. Can yall confirm if you'd expect a different value from `describe-create-account-status`? Perhaps something like... "READY" instead of "SUCCEEDED"?

https://github.com/hashicorp/terraform-provider-aws/blob/f9692d98fc5b64d97366424e3870acf5524e79e7/internal/service/organizations/account.go#L178

https://github.com/hashicorp/terraform-provider-aws/blob/f9692d98fc5b64d97366424e3870acf5524e79e7/internal/service/organizations/account.go#L440-L442

**Comment by ChristophShyper** [comment]
@drewmullen  That's the actual output on the AWS side - they never explicitly say when the account (initial IAM roles/user, VPC, etc.) is ready to handle Terraform provider calls. From  my years of experience, the most of the issues were caused by VPC and IAM services before 7 minutes of the account's creation.

**I've talked about this** with our **AWS TAM**  team  years ago. And I was even included in the organizations' development process projects. But the only progress I was able to talk about was to change of root emails' access that I had no access to.
